lib/internal/process.js

```diff
@@ -22,6 +23,46 @@ const assert = process.assert = function(x, msg) {
 };
 
 
+// Set up the process.cpuUsage() function.
+function setup_cpuUsage() {
+  // Get the native function, which will be replaced with a JS version.
+  const _cpuUsage = process.cpuUsage;
+
+  // Create the argument array that will be passed to the native function.
+  const cpuValues = new Float64Array(2);
+
+  // Replace the native function with the JS version that calls the native
+  // function.
+  process.cpuUsage = function cpuUsage(prevValue) {
+    // Call the native function to get the current values
+    if (_cpuUsage(cpuValues) !== 0) {
+      throw new Error('unable to obtain CPU usage time');
+    }
+
+    // If no previous value passed in, return current value.
+    if (prevValue == null) return {
+      user: cpuValues[0],
+      system: cpuValues[1]
+    };
+
+    // If a previous value was passed in, ensure it has the correct shape.
+    if (!Number.isFinite(prevValue.user)) {
```

Comment:
* `Number.isInteger()` - that's checking an implementation detail.  If for some reason we get nanosecond resolution later, with a different implementation API (other than libuv's rusage fn), then that constraint no longer holds.

* `> Number.MAX_SAFE_INTEGER` check - One of the advantages of using floats for these values is so that we don't have to worry about such things.  The # of microseconds that fit in MAX_SAFE_INTEGER is ~250 years worth, making it just on the cusp of ridiculous that we'd overflow that, but "who would ever need more than 640KB of RAM?" :-)